From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 3 Oct 2019 12:39:15 -0700
Subject: [PATCH] Switch to just building the traefik binary

Add install target as well.

2020-07-21 update: build a statically-linked PIE binary

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 Makefile | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 02fc779..354085f 100644
--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,13 @@ PRE_TARGET ?= build-dev-image
 
 PLATFORM_URL := $(if $(PLATFORM_URL),$(PLATFORM_URL),"https://pilot.traefik.io")
 
-default: binary
+default:
+	go build -mod=vendor -buildmode=pie -v -ldflags '-linkmode external -extldflags "-static-pie"' -tags netgo,osusergo
+
+install:
+	install -m 0755 -D traefik $(DESTDIR)/usr/bin/traefik
+	install -m 0644 -D contrib/systemd/traefik.service $(DESTDIR)/usr/lib/systemd/system/traefik.service
+	install -m 0644 -D traefik.sample.toml $(DESTDIR)/usr/share/traefik/traefik.sample.toml
 
 ## Build Dev Docker image
 build-dev-image: dist
